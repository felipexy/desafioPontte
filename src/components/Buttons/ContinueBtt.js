import React, { useState } from 'react';
import './ContinueBtt.scss';


const ContinueBtt = ({}) => {

    const [valid, setValid] = useState(false);

    function handleClick(){

    }

    return (
        <div className="flexContainer">
        <button className="continueBtt"  onClick={handleClick}><h2 className="content">Gostei, continuar</h2></button>  
        </div>
    );
};

export default ContinueBtt;