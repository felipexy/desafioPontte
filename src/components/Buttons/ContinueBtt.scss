.buttonContainer {
  display: flex;
}

.buttonContainer__continueBtt {
  border-radius: 50px;
  width: 35%;
  height: 6vw;
  margin-top: 4%;
  margin-right: auto;
  margin-left: auto;
  background: linear-gradient(
    90deg,
    rgba(130, 0, 131, 1) 10%,
    rgba(14, 0, 153, 1) 100%
  );
}

.buttonContainer__continueBtt:hover {
  background: linear-gradient(
    30deg,
    rgba(130, 0, 131, 1) 10%,
    rgba(14, 0, 153, 1) 100%
  );
  cursor: pointer;
  .content {
    color: white;
  }
}

.buttonContainer__continueBtt--NotValid {
  border-radius: 50px;
  width: 35%;
  height: 6vw;
  margin-top: 4%;
  margin-right: auto;
  margin-left: auto;
  background: gray;
  pointer-events: none;
  .content {
    font-weight: bold;
    margin-right: auto;
    margin-left: auto;
    margin-top: 5%;
    color: black;
    font-size: 1.7vw;
  }
}

.buttonContainer__continueBtt:focus {
  outline: 0 !important;
}

.buttonContainer__continueBtt__content {
  font-weight: bold;
  margin-right: auto;
  margin-left: auto;
  margin-top: 5%;
  color: white;
  font-size: 1.7vw;
}
