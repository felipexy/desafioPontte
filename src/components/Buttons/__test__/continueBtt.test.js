import React from 'react';
import ReactDOM from 'react-dom';
import ContinueBtt from '../ContinueBtt';
import { render, cleanup, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
describe('Tests for ContinueBtt component', () => {
  it('Click Button + Render', async () => {
    const { getByTestId } = render(<ContinueBtt />);
    const ctnBtt = await waitFor(() => getByTestId('contBtt'));
    fireEvent.click(ctnBtt);
  });

  it('Matches snapshot (notValid)', () => {
    const tree = renderer.create(<ContinueBtt isValid={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Matches snapshot (Valid)', () => {
    const tree = renderer.create(<ContinueBtt isValid />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
