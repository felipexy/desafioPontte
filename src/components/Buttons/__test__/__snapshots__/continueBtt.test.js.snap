// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ContinueBtt component Matches snapshot (Valid) 1`] = `
<div
  className="buttonContainer"
>
  <button
    className="buttonContainer__continueBtt"
    data-testid="contBtt"
    onClick={[Function]}
    type="submit"
  >
    <h2
      className="buttonContainer__continueBtt__content"
    >
      Gostei, continuar
    </h2>
  </button>
</div>
`;

exports[`Tests for ContinueBtt component Matches snapshot (notValid) 1`] = `
<div
  className="buttonContainer"
>
  <button
    className="buttonContainer__continueBtt--NotValid"
    data-testid="contBtt"
    onClick={[Function]}
    type="submit"
  >
    <h2
      className="content"
    >
      Gostei, continuar
    </h2>
  </button>
</div>
`;
